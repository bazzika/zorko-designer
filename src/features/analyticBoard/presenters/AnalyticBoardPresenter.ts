import {BaseStatePresenter} from '../../../core/BaseStatePresenter';
import {AnalyticBoardState} from './AnalyticBoardState';

export class AnalyticBoardPresenter extends BaseStatePresenter<AnalyticBoardState> {
  static create(state?: AnalyticBoardState) {
    return new AnalyticBoardPresenter(state);
  }

  getDefaultState(): AnalyticBoardState {
    return {
      mainSpecId: '',
      encodingChannels: []
    };
  }

  setMainSpec(id: string) {
    this.state.mainSpecId = id;
    return this;
  }

  getMainSpec() {
    return this.state.mainSpecId;
  }

  setChannels(channels: string[]) {
    this.state.encodingChannels = channels;
    return this;
  }

  getChannels() {
    return this.state.encodingChannels;
  }
}
