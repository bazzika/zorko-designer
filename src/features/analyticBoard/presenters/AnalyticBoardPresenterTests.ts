import {AnalyticBoardPresenter} from './AnalyticBoardPresenter';

describe('AnalyticBoardPresenter', () => {
  let presenter: AnalyticBoardPresenter;

  beforeEach(() => {
    presenter = AnalyticBoardPresenter.create();
  });

  it('creates with default state', () => {
    expect(AnalyticBoardPresenter.create().toState()).toMatchSnapshot();
  });

  it('sets main spec id', () => {
    expect(presenter.setMainSpec('123').toState()).toMatchSnapshot();
  });

  it('gets main spec id', () => {
    expect(presenter.setMainSpec('234').getMainSpec()).toEqual('234');
  });

  it('sets selected channels', () => {
    expect(presenter.setChannels(['bar/x', 'bar/y', 'bar/z']).toState()).toMatchSnapshot();
  });

  it('gets selected channels', () => {
    expect(presenter.setChannels(['bar/x', 'bar/y', 'bar/z']).getChannels()).toEqual([
      'bar/x',
      'bar/y',
      'bar/z'
    ]);
  });
});
