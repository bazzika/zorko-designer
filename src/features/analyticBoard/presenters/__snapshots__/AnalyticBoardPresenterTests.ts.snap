// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyticBoardPresenter creates with default state 1`] = `
Object {
  "encodingChannels": Array [],
  "mainSpecId": "",
}
`;

exports[`AnalyticBoardPresenter sets main spec id 1`] = `
Object {
  "encodingChannels": Array [],
  "mainSpecId": "123",
}
`;

exports[`AnalyticBoardPresenter sets selected channels 1`] = `
Object {
  "encodingChannels": Array [
    "x",
    "y",
    "z",
  ],
  "mainSpecId": "",
}
`;
